app/src/org/commcare/android/resource/analytics/UpdateStatPersistence.java

```diff
@@ -0,0 +1,69 @@
+package org.commcare.android.resource.analytics;
+
+import android.content.SharedPreferences;
+import android.util.Log;
+
+import org.commcare.dalvik.application.CommCareApp;
+
+import java.io.IOException;
+
+/**
+ * Saves and loads stats associated with an app update job to shared preferences.
+ *
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+public class UpdateStatPersistence {
```

Comment:
In a very uncharacteristic move, I might argue that this extra class adds unnecessary complexity, given that there are already so many different classes surrounding creating and logging the update statistics. These 3 methods might do fine as just static methods of the UpdateStats class.